### ISSUE TITLE
Table Transpose

### ISSUE THREAD

**Issue by CzechJiri** [metabase/metabase#3769]
ideally I would like metabase to be able to mimi excel's "paste as transpose" feature and swap rows for columns, please note this is not the same as "pivot table" feature introduced in 0.19, this is just plain transposing

For example if I have

```
DATE      AMOUNT1      AMOUNT2     AMOUNT3     AMOUNT4
201601         10           20          30          40
201602         50           60          70          80
```

the transpose would look like

```                
            201601           201602
AMOUNT1         10               50
AMOUNT2         20               60  
AMOUNT3         30               70 
AMOUNT4         40               80
```

**Comment by thoughtgap** [comment]
+1!

**Comment by tlrobinson** [comment]
Related to #2618.

Out of curiosity, how are you producing a query result like this?

I think we may want this (and pivoting, i.e. #2473) to happen as part of the query itself rather than a display setting, so CSV downloads, Pulses, etc, match.

We'd probably also want to resolve #664 first, or limit the number of rows that can be transposed.

**Comment by CzechJiri** [comment]
I use postgres 9.4

the first query is simple `SELECT myDate, sum(....), sum(....), sum(....) FROM .. GROUP BY 1`

The second query is a bit twisted monkey, I first have to use `unnest(array[...` on top of the result, then use `crosstab`

**Comment by tlrobinson** [comment]
The first one will also be possible using the query builder once we add multiple aggregation support (#3364) so we might want to tackle this as part of it.

Still not sure whether this should be client or server side. You'd expect the CSV download to match what you see in the table visualization, but there are already cases where that's not the case (pivot table, hiding / re-ordering columns).

**Comment by CzechJiri** [comment]
Any thoughts? This is very common feature in many analytical tools. 

I will buy a beer for everybody in your office just to get this implemented! How is that for an upvote

**Comment by alberthkcheng** [comment]
+1 for this feature. An checkbox display option to allow table transpose would be very helpful because those spreadsheet user tends to read thing horizontally, and this feature is very important step move people away from spreadsheet to METABASE.

It is very hard to do with `crosstab` in Postgres because the column (row) name can be dynamic.
In Python / R it is easy but not for SQL.

**Comment by dannyeuu** [comment]
@tlrobinson @flamber This has already implemented, right?!

**Comment by flamber** [comment]
@dannyeuu No, this is full transpose, whereas the current Pivot is just two groupings and a metric.
Unless I'm misreading the request.

**Comment by willtoth** [comment]
I posted some thoughts on what this could look like as part of the UI and another use case in this issue, I believe we are trying to achieve the same thing https://github.com/metabase/metabase/issues/12800

**Comment by tannercrook** [comment]
Domo has a powerful Transpose option on their tables to produce amazing funnels. I have successfully recreated this in SQL and Metabase but obviously doesn't have filter capabilities that work well and the ability to format rows. In Domo, the feature must transpose post query as you can set a currency row different from a percentage row. The columns should also be able to be formattable. Basically a standard table, but the first column row values get transposed to column headers. Could see the table visualization have a 'Transpose' option that does this. At least that is how Domo does it.

This feature would be a massive for Metabase. It is the single biggest thing needed in my opinion to keep people from constantly pulling into Excel to build these funnels. 

For example, spend should be formatted as currency and the number of users stay an integer. The columns (months) should also be able to be formatted properly.
<img width="1299" alt="Screen Shot 2022-09-13 at 11 40 20 PM" src="https://user-images.githubusercontent.com/6598028/190069443-55e24622-b3c6-43a3-a11c-0ef726512e93.png">

**Comment by e-gaulue** [comment]
If easier, couldn't we just have a new visualization with just the columns in lines. Filter (and all) will stay the same in the left column instead of top line?

It would already be good to show :
```SELECT TOP 1 Name, Address, ZipCode, City, Country, Phone FROM ...```
as
```
Name:		Paul
Address:	34 Stairway to heaven
ZipCode:	A1998
City:		Gomorrhe
Country:	Hell
Phone:		485 9384 384
```
in a dashboard.

**Comment by paoliniluis** [comment]
there's a new "detail" visualization in v46 that should do what you need

**Comment by jamesgreenblue** [comment]
@paoliniluis v46 has arrived and I can't see how the detail visualisation solves the transpose issue described by @CzechJiri unless I'm missing something obvious – the detail visualisation shows records one at a time?

**Comment by mmoole** [comment]
I really wonder what is taking this 'feature' so long? As a user, I would simply have a checkbox to swap rows and columns, and just the table-visualization should change. Isn't that all that is needed in essence?

**Comment by paoliniluis** [comment]
@mmoole we only count thumbs up

**Comment by mistakenot** [comment]
👍 this would solve a need for us.

**Comment by ZiadEK8** [comment]
Is there any chance that this feature is going to be made?

**Comment by rrr2rrr** [comment]
Better to have Pivot + Unpivot
Transpose = Unpivot -> Pivot

Pivot for PostgreSQL https://github.com/dverite/postgresql-functions/blob/master/pivots/dynamic-pivot-refcursor.sql
Unpivot can be done using the UNION. 

**Let's start with Unpivot only!**
Pivoting can be done manually column-by-column using the aggregates
In Aggregation string_agg can be added, that will be also helpful

**Comment by naduniprabhashwari** [comment]
Any update on this issue.
This will solve a lot of issues.

**Comment by longcao-cofano** [comment]
This would also solve our issues.
